components/ui/dropdown.vue

```diff
@@ -1,16 +1,33 @@
 <template>
-  <div :class="isOpen && 'open'" class="dropdown-shell">
+  <div :class="{open: isOpen}" class="dropdown-shell">
     <button
       @click="toggle"
       v-ripple="'rgba(56, 120, 0, .25)'"
       type="button"
       class="btn handle"
     >
-      <slot name="opener" />
+      <slot name="label">
+        {{ label }}
+      </slot>
       <img v-if="chevron" src="/images/events/chevron-down.svg" class="chevron" />
     </button>
-    <div :class="right && 'right-edge'" class="dropdown">
-      <slot />
+    <div :class="{'right-edge': right}" class="dropdown">
+      <div class="relative-wrapper">
+        <button @click="isOpen = false" class="btn normal round close">
+          <img src="/images/ui-kit/x.svg" />
+        </button>
+        <div
+          v-for="(option, i) in options"
+          :key="option.value"
```

Comment:
Dropdown is not only for the lists. There are could be just anything. Slot would work better

Reply:
That's why there is a slot inside it to customize how the item looks like. This provides a default interface to simplify the general case

Reply:
Also, making the entire dropdown content a slot would seem like taking too much repeated work outside of the component, like the dropdown saying "it's not my job" ))